### PR TITLE
Add more tests and clean up some code

### PR DESCRIPTION
**Pull request by protocol7**

### CODE REVIEW

File: src/test/java/com/spotify/folsom/client/OpCodeTest.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+package com.spotify.folsom.client;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class OpCodeTest {
+
+  @Test
+  public void testGetKind() throws Exception {
```
Comment (rschildmeijer) [inline src/test/java/com/spotify/folsom/client/OpCodeTest.java:25]: would it make sense to use an EnumSet, created using EnumSet.allOf() instead.
The advantage is less code + don't have to amend test if we add OpCodes.

File: src/test/java/com/spotify/folsom/transcoder/StringTranscoderTest.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+import org.junit.Test;
+
+import com.google.common.base.Charsets;
+
+public class StringTranscoderTest {
+
+  private static final String STRING = "hällo wörld";
```
Comment (rschildmeijer) [inline src/test/java/com/spotify/folsom/transcoder/StringTranscoderTest.java:27]: sure you didn't mean 'timo räisänen' ?

**Comment by rschildmeijer** [comment]
one minor comment otherwise +1

File: src/test/java/com/spotify/folsom/GetResultTest.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+  @Test
+  public void testEqualsHashCode() {
+    GetResult<Long> result1 = GetResult.success(123L, 456);
+    GetResult<Long> result2 = GetResult.success(123L, 456);
+    GetResult<Long> result3 = GetResult.success(999L, 999);
+    GetResult<Long> result4 = GetResult.success(123L, 999);
+    GetResult<Long> result5 = GetResult.success(999L, 456);
+
+    assertTrue(result1.equals(result1));
```
Comment (spkrka) [inline src/test/java/com/spotify/folsom/GetResultTest.java:38]: can't we use assertEquals instead? (though maybe that would make the assertFalse tests stand out more...)

File: src/test/java/com/spotify/folsom/client/UtilsTest.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  @Test(expected=IllegalArgumentException.class)
+  public void testValidateKeyInvalidChars() throws Exception {
+    Utils.validateKey("hällo");
+  }
+
+  @Test(expected=IllegalArgumentException.class)
+  public void testValidateKeyWithSpace() throws Exception {
+    Utils.validateKey("hello world");
+  }
+}
```
Comment (spkrka) [inline src/test/java/com/spotify/folsom/client/UtilsTest.java:43]: no newline at end of file - i think checkstyle was supposed to catch this.

**Comment by spkrka** [comment]
:+1:, i had some comments but nothing major

File: src/test/java/com/spotify/folsom/GetResultTest.java (hunk 3 above)
Comment (protocol7) [inline src/test/java/com/spotify/folsom/GetResultTest.java:38]: The reason I didn't use it was that that's not really the semantics of assertEquals, as that's supposed to compare the actual value with an expected value

File: src/test/java/com/spotify/folsom/client/OpCodeTest.java (hunk 1 above)
Comment (protocol7) [inline src/test/java/com/spotify/folsom/client/OpCodeTest.java:25]: Not sure how that would help with figuring out the kind?

File: src/test/java/com/spotify/folsom/client/UtilsTest.java (hunk 4 above)
Comment (protocol7) [inline src/test/java/com/spotify/folsom/client/UtilsTest.java:43]: Yeah, I noticed the same in another PR, so I guess not

**Comment by protocol7** [comment]
I'll go ahead and merge this, we can fix the comments later if we like

File: src/test/java/com/spotify/folsom/client/OpCodeTest.java (hunk 1 above)
Comment (rschildmeijer) [inline src/test/java/com/spotify/folsom/client/OpCodeTest.java:25]: Oh, OpCode is not an enum. So that wouldn't work

**Pull request merged**